File: app/src/main/java/com/jakubwyc/mysecretvault/mapper/RealmTreasureMapper.java
package com.jakubwyc.mysecretvault.mapper;

import com.jakubwyc.mysecretvault.model.Treasure;
import com.jakubwyc.mysecretvault.repository.RealmTreasure;


public class RealmTreasureMapper implements Mapper<RealmTreasure, Treasure> {

    @Override
    public Treasure map(RealmTreasure data) {
        return new Treasure(data.text, data.date);
    }
}

File: app/src/main/java/com/jakubwyc/mysecretvault/model/mapper/RealmTreasureMapper.kt
package com.jakubwyc.mysecretvault.model.mapper

import com.jakubwyc.mysecretvault.model.Treasure
import com.jakubwyc.mysecretvault.repository.RealmTreasure


class RealmTreasureMapper : Mapper<RealmTreasure, Treasure> {
    override fun map(data: RealmTreasure) = Treasure(data.text, data.date)
}
